src/rust/src/wallet.rs

```diff
@@ -58,22 +59,26 @@ pub struct InPoint {
 pub struct DecryptedNote {
     note: Note,
     memo: [u8; 512],
-    /// The position of the note's commitment within the global Merkle tree.
-    position: Option<Position>,
 }
 
-/// A data structure tracking the note data that was decrypted from a single transaction,
-/// along with the height at which that transaction was discovered in the chain.
+/// A data structure tracking the note data that was decrypted from a single transaction.
 #[derive(Debug, Clone)]
 pub struct TxNotes {
-    /// The block height containing the transaction from which these
-    /// notes were decrypted.
-    tx_height: Option<BlockHeight>,
     /// A map from the index of the Orchard action from which this note
     /// was decrypted to the decrypted note value.
     decrypted_notes: BTreeMap<usize, DecryptedNote>,
 }
 
+/// A data structure chain position information for a single transaction.
+#[derive(Clone, Debug, Default)]
+struct NotePositions {
+    /// The block height containing the transaction (if mined).
+    tx_height: Option<BlockHeight>,
+    /// A map from the index of an Orchard action tracked by this wallet, to the position
+    /// of the note's commitment within the global Merkle tree.
```

Comment:
Which note? @yingtong and I think it's the output note, but it's ambiguous.

Reply:
It's necessarily the output note, because we don't have commitments for spent notes within the transaction.

Reply:
```suggestion
    /// of the output note's commitment within the global Merkle tree.
```